packages/chip-input/src/ux-choice-container-attribute.ts

```diff
@@ -0,0 +1,105 @@
+import { UxChoiceAttribute } from './ux-choice-attribute';
+import { customAttribute, bindingMode, bindable, inject } from 'aurelia-framework';
+
+@customAttribute('ux-choice-container', bindingMode.twoWay)
+@inject(Element)
+export class UxChoiceContainerAttribute {
+
+  @bindable({defaultBindingMode: bindingMode.twoWay, primaryProperty: true}) public value: any | any[];
+  @bindable public multiple: boolean | 'auto' = 'auto';
+
+  public isMultiple: boolean = false;
+  private choices: Array<UxChoiceAttribute> = [];
+  private processClick: (event: Event) => void;
+
+  constructor(private element: Element) {
+  }
+
+  public bind() {
+    this.multipleChanged();
+    this.valueChanged(this.value);
+  }
+
+  public multipleChanged() {
+    if (this.multiple === 'auto') {
+      this.isMultiple = Array.isArray(this.value) ? true : false;
+    } else if (typeof this.multiple === 'boolean') {
+      this.isMultiple = this.multiple;
+    } else {
+      this.isMultiple = this.multiple === 'multiple' || this.multiple === 'true';
+    }
+    this.element.classList.toggle('ux-choice-container--multiple', this.isMultiple);
+  }
+
+  public attached() {
+    this.element.classList.add('ux-choice-container');
+    this.processClick = (event: Event) => {
+      if (event.target instanceof Element) {
+        const choiceElement: any = event.target.closest('.ux-choice');
+        if (
+          choiceElement !== null &&
+          choiceElement.au !== undefined &&
+          choiceElement.au['ux-choice'] !== undefined &&
+          choiceElement.au['ux-choice'].viewModel instanceof UxChoiceAttribute) {
+          const choice: UxChoiceAttribute = choiceElement.au['ux-choice'].viewModel;
+          this.toggleValue(choice.value);
+        }
+      }
+    };
+    this.element.addEventListener('click', this.processClick, false);
+  }
+
+  public detached() {
+    this.element.removeEventListener('click', this.processClick, false);
+  }
+
+  public registerChoice(choice: UxChoiceAttribute) {
+    this.choices.push(choice);
+    this.processValue();
```

Comment:
I feel this is a bit too eager when processing. Does it make sense to delay for 1 tick before we process the value? Imagine the scenarios where an array with 50 choices are given based on some key strokes

Reply:
This is not critical right now, but we can keep this in mind in case we need to tweak for perf

Reply:
Good point. I would be happy to implement this delay. The only way I currently know how to do it is with a `setTimeout(/*..*/, 0)` that is cleared each time another iteration require the same processing. Are there any other smarter way to do it (without `setTimeout` ?). The `TaskQueue` doesn't have an API for such scenarios ?

Reply:
`setTimeout` would be a bit too much in terms of timing. We can use `microTaskQueue` for this:
```ts
taskQueue.queueMicroTask(() => this.processValue());
```
And like the API `handleEvent`, we can use an object with `call` method instead of a function. So we can add a method `call` to choice-container class:
```ts
call() {
  this.processValue();
}
```
then we can just do:
```ts
taskQueue.queueMicroTask(this);
```

Reply:
So when we pass an object with `call` method several times in a row and hasn't been called yet, the task queue only keep on reference to the micro task to call ?

Reply:
> the task queue only keep on reference to the micro task to call ?
yes, it will call only once, in the next micro task run. which means even a 1000 register calls will still result into only 1 `processValue` call

Reply:
Wow amazing !!

Reply:
Hum... just tried the taskQueue trick but it still call the `call` as many times as it has been registered ... I have tried `queueMicroTask(this)` and `queueTask(this)`

Reply:
If I look at the TaskQueue source I think it will [register each call here](    this.microTaskQueue.push(task);) event if a `Task` interface is passed on.

Reply:
ugh, sorry i forgot that we need to implement the guard ourself (been doing this for a long time).
so it would look like this:
```ts
isQueued: boolean = false;

call() {
  this.isQueued = false;
}

registerChoice(choice) {
  this.choices.push(choice);
  if (!this.isQueued) {
    this.isQueued = true;
    this.microTaskQueue.queueMicroTask(this);
  }
}
```

Reply:
makes sense! Thanks !